Biniverse/server-test
# [BE]: Sign in

# Pull Request: Implement User Login Functionality

## Description
This pull request introduces a new user login feature in our Express application. It includes the `loginUser` function, which handles login requests by validating the provided credentials against predefined constants.

## Changes Made
- Added the `loginUser` function to handle user login via username or email.
- Implemented validation logic to check credentials against hardcoded values.
- Included appropriate HTTP response status codes and messages for success and failure cases.

## Related Issues
- Closes #42 (User authentication feature)

## Type of Change

## How to Test
1. Start the Express server.
2. Send a POST request to the `/login` endpoint with a JSON body containing either a `username` or `email`, along with `password` and an `app-key`.
3. Verify that valid credentials return a 200 status with a success message, while invalid credentials return a 401 status with an error message.

## Checklist
- [x] Code is written and works as expected.
- [x] Linting and formatting are consistent with project guidelines.

## Additional Notes
- This implementation uses hardcoded values for demonstration purposes. Consider implementing a database for user credentials in future updates.
